scipy/stats/stats.py

```diff
@@ -3454,19 +3470,127 @@ def ttest_ind(a, b, axis=0, equal_var=True):
-
-    if equal_var:
-        df, denom = _equal_var_ttest_denom(v1, n1, v2, n2)
+    random_state = check_random_state(random_state)
+        
+    if permutations is not None:
+        mat = np.concatenate((a, b), axis=axis)
+        cats = np.hstack((np.zeros(a.shape[axis]), np.ones(b.shape[axis])))
+        t_stat, pvalues = _permutation_ttest(mat, cats,
+                                             axis=axis,
+                                             equal_var=equal_var,
+                                             permutations=permutations,
+                                             random_state=random_state)
+        return t_stat, pvalues
     else:
-        df, denom = _unequal_var_ttest_denom(v1, n1, v2, n2)
+        v1 = np.var(a, axis, ddof=1)
+        v2 = np.var(b, axis, ddof=1)
+        n1 = a.shape[axis]
+        n2 = b.shape[axis]
+
+        if equal_var:
+            df, denom = _equal_var_ttest_denom(v1, n1, v2, n2)
+        else:
+            df, denom = _unequal_var_ttest_denom(v1, n1, v2, n2)
+        return _ttest_ind_from_stats(np.mean(a, axis),
+                                     np.mean(b, axis),
+                                     denom, df)
+
+
+def _init_summation_index(cats):
+    """
+    Creates a matrix filled with category permutations
+    
+    cats: numpy.array
+       List of class assignments
+    """
+    c = len(cats)
+    num_cats = len(np.unique(cats))  # Number of distinct categories
+    copy_cats = copy.deepcopy(cats)
+    perms = np.array(np.zeros((c, num_cats), dtype=cats.dtype))
+    for i in range(num_cats):
+        perms[:,i] = (copy_cats == i).astype(cats.dtype)
+    return perms
+
+
+def _permutation_ttest(mat, cats, axis=0, permutations=10000, equal_var=True, random_state=None):
+    """
+    Calculates the T-test for the means of TWO INDEPENDENT samples of scores
+    using permutation methods
 
-    return _ttest_ind_from_stats(np.mean(a, axis),
-                                 np.mean(b, axis),
-                                 denom, df)
+    This test is an equivalent to scipy.stats.ttest_ind, except it doesn't require the
+    normality assumption since it uses a permutation test .
+
+    Parameters
+    ----------
+    mat : array_like
+        This contains all of the data values for both groups.
+    cats: array_like
+        This array must be 1 dimensional and have the same size as
+        the length of the specified axis.  This encodes information
+        about which category each element of the mat array belongs to
+    axis : int, optional
+        Axis can equal None (ravel array first), or an integer (the axis
+        over which to operate on a and b).
+    permutations: int
+        Number of permutations used to calculate p-value
+    equal_var: bool
+        If false, a Welch's t-test is conducted.  Otherwise,
+        a ordinary t-test is conducted
+    random_state : int or RandomState
+        Pseudo number generator state used for random sampling.
```

Comment:
Indicate that it only is used if the p-value is computed using permutation test.